libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/validator/VerifiableCredentialValidator.java

```diff
@@ -140,6 +154,28 @@ private void validateSignature(SignedJWT verifiableCredential, ECKey signingKey)
         }
     }
 
+    private JWSVerifier getVerifier(EncryptionAlgorithm signingAlgorithm, String signingKey)
+            throws ParseException, JOSEException {
+        return switch (signingAlgorithm) {
+            case EC -> new ECDSAVerifier(ECKey.parse(signingKey));
+            case RSA -> new RSASSAVerifier(RSAKey.parse(signingKey));
```

Comment:
Digging a bit further into our libraries, it looks like you might be able to do this completely automatically.

Have a look at `DefaultJWSVerifierFactory.createJWSVerifier()` and `JWK.parse()`

Reply:
I saw JWK.parse, and am now updating to use the key to decide. Will also look at createJWSVerifier

Reply:
Trying to play around with createJWSVerifier but it feels like it may not be as much of a simplification as wanted compared to the new version of the getVerifier method.

```
            var signingKeyJws = JWSObject.parse(signingKey);
            var keyFactory = KeyFactory.getInstance(signingKeyJws.getHeader().getAlgorithm().toString());
            keyFactory.generatePublic(new PKCS8EncodedKeySpec(signingKey.getBytes())
            );

            new DefaultJWSVerifierFactory().createJWSVerifier(signingKeyJws.getHeader(), Key);

            var formattedVc = EC.equals(signingKeyJws.getHeader().getAlgorithm()) ? transcodeSignatureIfDerFormat(vc) : vc;

```